### PR TITLE
fix(FAN-41): catch the EnrollmentNotAllowed and InvalidEnrollmentAttribute

### PR DESCRIPTION
**Pull request by MaferMazu**
## Description

This PR catches EnrollmentNotAllowed and InvalidEnrollmentAttribute exceptions in favor of returning a status 400 bad request. This fixes the 500 error when you make a post request to the /api/enrollment/v1/enrollment.

## How to test
- Have an environment with this version of edx-platform.
- Have the filter configured and a course with the proper "other course settings" as we explained [in the FAN-41 PR](https://github.com/fccn/nau-openedx-extensions/pull/5).
- With a user without an allowed domain, make a post request to /api/enrollment/v1/enrollment with `{"course_details":{"course_id": "<your_course_id>"}}`, and you are going to get an error 400 instead of 500.

### Before
Screenshot from 2022-12-14 16-59-19
Response in /api/enrollment/v1/enrollment
Screenshot from 2022-12-14 16-58-50
Logs

### After
Screenshot from 2022-12-14 17-08-31
Response in /api/enrollment/v1/enrollment
Screenshot from 2022-12-14 17-07-31
Logs

> Note: If you want to change the message, you need to edit [this](https://github.com/fccn/nau-openedx-extensions/blob/nau/lilac.master/nau_openedx_extensions/filters/pipeline.py#L42) and its translation [here](https://github.com/fccn/nau-openedx-extensions/blob/nau/lilac.master/nau_openedx_extensions/locale/pt_PT/LC_MESSAGES/django.po#L110).

### CODE REVIEW

**Review by felipemontoya — APPROVED** [review]
This looks exactly as we discussed in the call.
The code is tested in a development environment and the screenshots are pasted here. I think we are good to merge. Just waiting for Ivo's approval

**Review by igobranco — APPROVED** [review]

**Pull request merged**
